tensorflow/contrib/verbs/rdma.cc

```diff
@@ -271,6 +271,10 @@ RdmaChannel::RdmaChannel(const RdmaAdapter* adapter, const string local_name,
     self_.lid = attr.lid;
     self_.qpn = qp_->qp_num;
     self_.psn = static_cast<uint32_t>(random::New64()) & 0xffffff;
+    union ibv_gid gid;
+    CHECK(!ibv_query_gid(adapter_->context_, (uint8_t) 1, 0, &gid)) << "Query gid";
```

Comment:
@shamoya - Thanks for taking a look at this patch. 
Regarding device selection, @bkovalev raised this issue to me. However, I have not had time to work on it. Feel free to send a PR.